client/my-sites/site-settings/controller.js

```diff
@@ -118,31 +115,53 @@ module.exports = {
 	},
 
 	deleteSite( context ) {
-		let site = sites.getSelectedSite();
+		const { dispatch, getState } = context.store;
+		const siteSlug = getSelectedSiteSlug( getState() );
+		const siteId = getSelectedSiteId( getState() );
 
-		if ( sites.initialized ) {
-			if ( ! canDeleteSite( site ) ) {
-				return page( '/settings/general/' + site.slug );
+		const redirectIfCantDelete = () => {
```

Comment:
@tyxla I've pushed [50e93a6](https://github.com/Automattic/wp-calypso/pull/13468/commits/50e93a67e584902f0f5dd8c804000968a627464f) which does something like what you proposed.